pypeit/core/coadd.py

```diff
@@ -549,24 +552,46 @@ def solve_poly_ratio(wave, flux, ivar, flux_ref, ivar_ref, norder, mask = None,
     # Determine an initial guess
     ratio = robust_median_ratio(flux, ivar, flux_ref, ivar_ref, mask=mask, mask_ref=mask_ref,
                                 ref_percentile=ref_percentile, max_factor=scale_max)
+    wave_min = wave.min()
+    wave_max = wave.max()
+
```

Comment:
Take a look at the changes made in my branch mosfire_oct20. Please test on your problem.

Reply:
I think mosfire_oct20 works great to fix the problems I was having. One weird thing, though, is when I test with real data the fit looks great, but I still get the same error message from `scipy.optimize.minimize`. However, when I look at nearby cuts through parameter space, the loss function looks smooth. I think this may truly be a numerical precision problem then, and we might want to switch to using a different optimization method in `poly_ratio_fitfunc`, like Powell e.g.